macros.in

```diff
@@ -245,6 +245,10 @@ Supplements:   (%{name} = %{version}-%{release} and langpacks-%{1})\
 #	Is ignored when SOURCE_DATE_EPOCH is not set.
 %clamp_mtime_to_source_date_epoch 0
 
+#	If true, make sure that timestamps in built rpms
+#	are not later than the build time of the package.
+%clamp_mtime_to_buildtime 0
```

Comment:
This means we now have two independent knobs for the same thing, and both cannot be right.
I'd rather have something like `%clamp_mtime` macro that accepts none/buildtime/source_date_epoch as possible values. `%clamp_mtime_to_source_date_epoch` can then be deprecated and routed to the new macro for now, for backwards compatibility.

Reply:
I have not verified if that works, but if one can use lua in the %{?_buildtime} macro to read and set environment variables and execute the date binary then we could do the logic in macros.

And thus also use %{?_mtime} as returning the epoch to set. If it is something like a false value that is not literal 0, the we would read it from disk like the current default. Not sure why someone would want to clamp it, instead of setting it, but if we need to we could have %{?_mtime_treatment} with 'set' or 'clamp'. I guess a callback to a macro for each file might be too slow? What do you think?